app/qml/Highlight.qml

```diff
@@ -148,11 +148,19 @@ Item {
         let objOwner = ( geometryType === 1 ? lineShapePath : polygonShapePath )
         let elements = ( geometryType === 1 ? newLineElements : newPolygonElements )
 
-        // move brush to the first point and start drawing from next point
-        elements.push( componentMoveTo.createObject( objOwner, { "x": data[ dataStartIndex ], "y": data[ dataStartIndex + 1 ] } ) )
-        for ( let i = dataStartIndex + 2; i < data.length; i += 2 )
+        // Create (multi) geometry for the highlight
+        let i = 0
+        let k = 0
+        while ( i < data.length )
         {
-          elements.push( componentLineTo.createObject( objOwner, { "x": data[ i ], "y": data[ i + 1 ] } ) )
+          let geomType = data[ i++ ];
+          let pointsCount = data[ i++ ];
+          elements.push( componentMoveTo.createObject( objOwner, { "x": data[ i ], "y": data[ i + 1 ] } ) )
+          for ( k = i; k < i + pointsCount * 2; k += 2 )
```

Comment:
it looks like we are adding the first point with both moveTo and lineTo if we start from `k = i`